src/bun.js/webcore/blob.zig

```diff
@@ -3423,12 +3431,192 @@ pub const Blob = struct {
+        usingnamespace bun.New(S3BlobStatTask);
+
+        pub fn onS3StatResolved(result: AWS.S3StatResult, this: *S3BlobStatTask) void {
+            defer this.deinit();
+            switch (result) {
+                .not_found => {
+                    this.promise.resolve(this.globalThis, .false);
+                },
+                .success => |stat| {
+                    if (this.blob.size == Blob.max_size) {
+                        this.blob.size = @truncate(stat.size);
+                    }
+                    this.promise.resolve(this.globalThis, .true);
+                },
+                .failure => |err| {
+                    const js_err = this.globalThis.createErrorInstance("{s}", .{err.message});
+                    js_err.put(this.globalThis, ZigString.static("code"), ZigString.init(err.code).toJS(this.globalThis));
+                    this.promise.rejectOnNextTick(this.globalThis, js_err);
+                },
+            }
+        }
+
+        pub fn init(globalThis: *JSC.JSGlobalObject, blob: *Blob, js_blob: JSValue) JSValue {
+            const this = S3BlobStatTask.new(.{
+                .globalThis = globalThis,
+                .blob = blob,
+                .promise = JSC.JSPromise.Strong.init(globalThis),
+                .strong_ref = JSC.Strong.create(js_blob, globalThis),
+            });
+            const promise = this.promise.value();
+            const env = this.globalThis.bunVM().bundler.env;
+            const credentials = env.getAWSCredentials();
+            const url = bun.URL.parse(this.blob.store.?.data.file.pathlike.path.slice());
+            this.poll_ref.ref(globalThis.bunVM());
+
+            credentials.s3Stat(url.s3Path(), @ptrCast(&S3BlobStatTask.onS3StatResolved), this, if (env.getHttpProxy(url)) |proxy| proxy.href else null);
+            return promise;
+        }
+
+        pub fn deinit(this: *S3BlobStatTask) void {
+            this.poll_ref.unrefOnNextTick(this.globalThis.bunVM());
+            this.strong_ref.deinit();
+            this.promise.deinit();
+            this.destroy();
+        }
+    };
+
+    pub fn getPresignUrl(this: *Blob, globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) bun.JSError!JSValue {
+        if (this.isS3()) {
+            const args = callframe.arguments_old(1);
+            var method: bun.http.Method = .GET;
+            var expires: usize = 86400; // 1 day
+            if (args.len == 1) {
+                const options = args.ptr[0];
+                if (options.isObject()) {
+                    if (try options.getTruthyComptime(globalThis, "method")) |method_| {
+                        method = Method.fromJS(globalThis, method_) orelse {
+                            return globalThis.throwInvalidArguments("method must be GET, PUT, DELETE or HEAD when using s3 protocol", .{});
+                        };
+                    }
+                    if (try options.getTruthyComptime(globalThis, "expiresIn")) |expires_| {
```

Comment:
```suggestion
                    if (try options.getOptional(i32, globalThis, "expiresIn")) |expires_| {
```

Reply:
i32 was not implemented in getOptional added one but is basically the same thing I was doing before.